src/discordtipbot/discordbot/premium_cmd.cr

```diff
@@ -0,0 +1,43 @@
+class PremiumCmd
+  def initialize(@tip : TipBot)
+  end
+
+  def call(msg, ctx)
+    client = ctx[Discord::Client]
+    cmd = ctx[Command].command
+    cache = client.cache.not_nil!
+
+    begin
+      method = cmd[1]? if guild_id = cmd[0]?.try &.to_u64
+    rescue
+    end
+    guild_id = cache.resolve_channel(msg.channel_id).guild_id.try &.to_u64 unless guild_id
+    return unless guild_id
+
+    @tip.clear_expired_premium
+
+    case method || cmd[0]?
+    when "status"
+    when "extend"
+      time_array = method ? cmd[2..3] : cmd[1..2]
+      @tip.extend_premium(Premium::Kind::Guild, guild_id, time(time_array))
+    end
+
+    human = ctx[ConfigMiddleware].get_premium_string(Premium::Kind::Guild, guild_id)
+    string = human ? "The guild has premium for another **#{human}**" : "The guild does **not** have premium"
+    client.create_message(msg.channel_id, string)
+    yield
+  end
+
+  private def time(time_array) : Time::Span
+    number, span = time_array
+    case span
+    when .starts_with?("minute") then number.to_i.minutes
```

Comment:
If I send a value for `"5 minutes"`, plurality is not handled, so it will be `0.seconds`.. I think most users will expect plurality to work.

Reply:
This isn't exposed to users in any way 😏

It's #starts_with? so it'll actually work with plural and singular

Reply:
Ah, right. Explicit plurality would be nice, but this is fine I guess.